components/sup/src/main.rs

```diff
@@ -153,31 +152,16 @@ fn config_from_args(subcommand: &str, sub_args: &ArgMatches) -> Result<()> {
         .unwrap_or(DEFAULT_LISTEN_SWIM)));
     config.set_gossip_listen(String::from(sub_args.value_of("listen-gossip")
         .unwrap_or(DEFAULT_LISTEN_GOSSIP)));
-
-    // NOTE: ip() returns an IpAddr, which we conveniently turn into a string
-    // via to_string().
     let default_gossip_ip = try!(ip()).to_string();
     let (gossip_ip, gossip_port) = try!(parse_ip_port_with_defaults(
                                         sub_args.value_of("listen-peer"),
                                         &default_gossip_ip,
                                         DEFAULT_GOSSIP_LISTEN_PORT));
-
-    debug!("Gossip IP = {}", &gossip_ip);
-    debug!("Gossip port = {}", &gossip_port);
     config.set_gossip_listen_ip(gossip_ip);
     config.set_gossip_listen_port(gossip_port);
-
-    let (sidecar_ip, sidecar_port) = try!(parse_ip_port_with_defaults(
-                                            sub_args.value_of("listen-http"),
-                                            DEFAULT_HTTP_LISTEN_IP,
-                                            DEFAULT_HTTP_LISTEN_PORT));
-
-    debug!("HTTP IP = {}", &sidecar_ip);
-    debug!("HTTP port = {}", &sidecar_port);
-
-    config.set_http_listen_ip(sidecar_ip);
-    config.set_http_listen_port(sidecar_port);
-
+    if let Some(addr_str) = sub_args.value_of("listen-http") {
```

Comment:
yeah - since this is stringy, I think my comment above about just taking `SocketAddr` rather than a specific v4 is the right thing, as we will then work transparently with v6.